### PR TITLE
Magento_Email: avoid using deprecated escape* methods from AbstractBlock

### PR DESCRIPTION
**Pull request by sergiynezbritskiy**
### Description (*)

Avoid using deprecated escape* methods from AbstractBlock

### Related Pull Requests

https://github.com/magento/magento2/pull/31610

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Please check and fix the failing unit tests

**Comment by sergeynezbritskiy** [comment]
@magento run all tests

File: app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php [hunk 1]
```diff
@@ -18,17 +20,23 @@ class SenderTest extends TestCase
     /**
      * @var MockObject|Sender
      */
-    protected $block;
+    private $block;
```
Comment (ihor-sviziev) [inline app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php:23]: Please fix phpdoc block here, it's not the mock

File: app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php [hunk 2]
```diff
@@ -18,17 +20,23 @@ class SenderTest extends TestCase
...
+
+    /**
+     * @var MockObject|Escaper
+     */
+    private $escaper;
```
Comment (ihor-sviziev) [inline app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php]:
```suggestion
    private $escaperMock;
```

File: app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php [hunk 3]
```diff
@@ -18,17 +20,23 @@ class SenderTest extends TestCase
...
     /**
      * Setup environment
      */
     protected function setUp(): void
     {
-        $this->block = $this->getMockBuilder(Sender::class)
-            ->disableOriginalConstructor()
-            ->setMethods(['escapeHtml'])
-            ->getMock();
+        $this->escaper = $this->createMock(Escaper::class);
```
Comment (ihor-sviziev) [inline app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php]:
```suggestion
        $this->escaperMock = $this->createMock(Escaper::class);
```

File: app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php [hunk 4]
```diff
@@ -18,17 +20,23 @@ class SenderTest extends TestCase
...
+        $context = $this->createMock(Context::class);
+        $context->method('getEscaper')->willReturn($this->escaper);
```
Comment (ihor-sviziev) [inline app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php]:
```suggestion
        $contextMock = $this->createMock(Context::class);
        $contextMock->method('getEscaper')->willReturn($this->escaperMock);
```

File: app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php [hunk 5]
```diff
@@ -18,17 +20,23 @@ class SenderTest extends TestCase
...
+
+        $this->block = new Sender($context);
```
Comment (ihor-sviziev) [inline app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php]:
```suggestion
        $this->block = new Sender($contextMock);
```

File: app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php [hunk 6]
```diff
@@ -37,8 +45,8 @@ protected function setUp(): void
     public function testRenderWithSenderNameAndEmail()
     {
         $templateSenderEmail = 'test';
-        $this->block->expects($this->any())->method('escapeHtml')->with($templateSenderEmail)
-            ->willReturn('test');
+        $this->escaper->expects($this->any())->method('escapeHtml')->with($templateSenderEmail)
```
Comment (ihor-sviziev) [inline app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php]:
```suggestion
        $this->escaperMock->expects($this->any())->method('escapeHtml')->with($templateSenderEmail)
```

File: app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php [hunk 7]
```diff
@@ -56,8 +64,8 @@ public function testRenderWithSenderNameAndEmail()
     public function testRenderWithNoSenderNameAndEmail()
     {
         $templateSenderEmail = '';
-        $this->block->expects($this->any())->method('escapeHtml')->with($templateSenderEmail)
-            ->willReturn('');
+        $this->escaper->expects($this->any())->method('escapeHtml')->with($templateSenderEmail)
```
Comment (ihor-sviziev) [inline app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php]:
```suggestion
        $this->escaperMock->expects($this->any())->method('escapeHtml')->with($templateSenderEmail)
```

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php (hunk 1 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php:23]: Fixed

File: app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php (hunk 5 above)
Comment (sergiynezbritskiy) [reply app/code/Magento/Email/Test/Unit/Block/Adminhtml/Template/Render/SenderTest.php]: Fixed

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by sergeynezbritskiy** [comment]
Hi @ihor-sviziev I've added the changes you requested, please proceed with the review of this PR
Regards

**Comment by sergeynezbritskiy** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Review by ihor-sviziev — APPROVED** [review]

**Pull request closed without merging**